### ISSUE TITLE
Reusable workflows referenced by path (which is valid) are not being allowed

### ISSUE THREAD

**Issue by jamiet-msm** [rhysd/actionlint#114]
At https://docs.github.com/en/actions/using-workflows/reusing-workflows#calling-a-reusable-workflow it is stated:

> You reference reusable workflow files using one of the following syntaxes:
> 
> * {owner}/{repo}/{path}/{filename}@{ref} for reusable workflows in public repositories.
> * ./{path}/{filename} for reusable workflows in the same repository.

I have just created a reusable workflow in the same repository as the caller workflow (so matching the second bullet point from above) and am calling it like so:

```yaml
dp_deploy_dev:
    uses: ./.github/workflows/deploy-asset.yml
```

actionlint is returning this error:

```
.github/workflows/ci.yml:378:11: reusable workflow call "./.github/workflows/deploy-asset.yml" at "uses" is not following the format "owner/repo/path/to/workflow.yml@ref". see https://docs.github.com/en/actions/learn-github-actions/reusing-workflows for more details [workflow-call]
    |
378 |     uses: ./.github/workflows/deploy-asset.yml
    |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
------
```

Given the documentation quoted above I'm pretty certain this is valid syntax and actionlint is raising a false positive. Would you agree?

**Comment by jamiet-msm** [comment]
Until this is fixed (or until you tell me that its a PEBCAK problem) Is there a way to ignore actionlint errors using an inline ignore statement?

**Comment by jamiet-msm** [comment]
I've temporarily gotten around this problem by specifying `VALIDATE_GITHUB_ACTIONS: false` in my super-linter job

**Comment by stabai** [comment]
This seems to be the code causing this lint failure:
https://github.com/rhysd/actionlint/blob/6abee944c9304acf6d6d62ab1e54fced80a3f5e9/rule_workflow_call.go#L57-L59

I think this should be an early return of `true` instead to fix the issue, but it's possible more validation than that would be appropriate.

**Comment by LeoColomb** [comment]
> since you can't expect any particular path structure

You do expect a particular path structure: `.github/workflows`, if I'm reading the docs correct 😊
https://docs.github.com/en/actions/using-workflows/reusing-workflows#creating-a-reusable-workflow

**Comment by stabai** [comment]
Whoops, you're right. It looks to me like `.github/workflows/` will also fail though, and that it actually needs to be `./.github/workflows/` (leaving off the . at the beginning causes the workflow to be ignored as invalid).

**Comment by stabai** [comment]
I've updated my PR to only allow that path structure, and also updated tests: https://github.com/rhysd/actionlint/pull/115

**Comment by akitson-degreed** [comment]
Workaround for now, to ignore this specific error:
`-ignore 'reusable workflow call ".+" at "uses" is not following the format ".+". see https://docs.github.com/en/actions/learn-github-actions/reusing-workflows for more details [workflow-call]'`

**Comment by stabai** [comment]
I just saw that https://github.com/rhysd/actionlint/pull/107 was also opened a few weeks ago, but has no comments from @rhysd :( Hopefully we get some movement on this.

Thank you @akitson-degreed for the description of the workaround!

**Comment by stabai** [comment]
Nice, 107 was merged and v1.6.9 has been released with that included! Thanks @rhysd!

**Comment by stabai** [comment]
I've also proposed https://github.com/github/super-linter/pull/2549 in super-linter to bump its version of actionlint to v1.6.9.

**Comment by rhysd** [comment]
Thanks for confirming that this issue was fixed. Closing.

**Comment by jamiet-msm** [comment]
PR to use actionlint 1.6.9 in super-linter has been merged https://github.com/github/super-linter/pull/2554 (8 days ago)
